host/3.0/buster/amd64/powershell/powershell6-core-tools.Dockerfile

```
@@ -0,0 +1,100 @@
+    # generate locale
+    && locale-gen && update-locale \
+    # remove powershell package
+    && rm /tmp/powershell.deb \
+    #
+    # Install Azure Functions and Azure CLI
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/azure-cli.list \
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-debian-$(lsb_release -cs)-prod $(lsb_release -cs) main" > /etc/apt/sources.list.d/dotnetdev.list \
+    && curl -sL https://packages.microsoft.com/keys/microsoft.asc | (OUT=$(apt-key add - 2>&1) || echo $OUT) \
+    && apt-get update \
+    && apt-get install -y azure-cli azure-functions-core-tools-3 \
+    #
+    # Create a non-root user to use if preferred - see https://aka.ms/vscode-remote/containers/non-root-user.
+    && groupadd --gid $USER_GID $USERNAME \
+    && useradd -s /bin/bash --uid $USER_UID --gid $USER_GID -m $USERNAME \
+    # [Optional] Add sudo support for the non-root user
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME\
+    && chmod 0440 /etc/sudoers.d/$USERNAME \
+    # intialize powershell module cache
+    && pwsh \
+        -NoLogo \
+        -NoProfile \
+        -Command " \
+        \$ErrorActionPreference = 'Stop' ; \
+        \$ProgressPreference = 'SilentlyContinue' ; \
+        while(!(Test-Path -Path \$env:PSModuleAnalysisCachePath)) {  \
+            Write-Host "'Waiting for $env:PSModuleAnalysisCachePath'" ; \
+            Start-Sleep -Seconds 6 ; \
+        }" \
+    # install AzModule
+    && pwsh \
+        -NoLogo \
+        -NoProfile \
+        -Command " \
+        Set-PSRepository PSGallery -InstallationPolicy Trusted; \
+        Install-Module -Name Az -AllowClobber -Scope AllUsers -Confirm:\$False -Force" \
+    # initialize powershell module cache
+    && pwsh -NoLogo -NoProfile -Command "try { IAmSureThisCommandDoesNotExist } catch { exit 0 }" \
```

I'm wondering why `exit 0`. Why not just ignore the exception and do nothing?

---

Done

---

@AnatoliB if we pass it, it return status code 1. so it fails pipeline. 

```
$ pwsh -NoLogo -NoProfile -Command "try { IAmSureThisCommandDoesNotExist } catch {  }"
$ echo $?
1
$ pwsh -NoLogo -NoProfile -Command "try { IAmSureThisCommandDoesNotExist } catch { exit 0 }"
$ echo $?
0
```

---

I see, you are right, thank you.